### PR TITLE
Charm default to app name

### PR DESCRIPTION
**Pull request by gtrkiller**
This PR aims to change the default value of the external_hostname config value to an empty string, and add code to make it default to the app name when the string is empty.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -91,6 +90,11 @@ def _mongodb_client_relation_changed(self, event: ops.framework.EventBase) -> No
         )
         if initial != self._stored.reldata["mongodb"]:
             self._configure_workload(event)
+    
+    def _check_external_hostname(self, hostname):
```
Comment (arturo-seijas) [inline src/charm.py:96]: The method name seems a bit ambiguous. I wouldn't expect it to return a value. Can you rename it?

**Review by merkata — APPROVED** [review]
Looks good! Approving.

**Review by arturo-seijas — APPROVED** [review]
LGTM

File: config.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ options:
   external_hostname:
     type: string
     description: "External hostname this gunicorn app should respond to (required)."
```
Comment (mthaddon) [inline config.yaml:16]: Can we update the description to say if left blank the deployed juju application name will be used?

File: src/charm.py [hunk 3]
```diff
@@ -92,6 +93,11 @@ def _mongodb_client_relation_changed(self, event: ops.framework.EventBase) -> No
         if initial != self._stored.reldata["mongodb"]:
             self._configure_workload(event)
 
+    def _assign_external_hostname(self, hostname) -> str:
```
Comment (mthaddon) [inline src/charm.py:96]: I think the name of `_get_external_hostname` would be more consistent with other methods in this charm. Also, you can access `self.config` here too, which would simplify calls to this function by not needing to pass a parameter, so I think that's a better option. Also, Can you include a docstring?

**Review by mthaddon — CHANGES_REQUESTED** [review]
Some comments inline, but can we also update the README - it currently says:
"""
Once the deployment has completed and the "gunicorn-k8s" workload state in
`juju status` has changed to "active" you can visit `http://gunicorn` in
a browser (assuming `gunicorn` resolves to the IP(s) of your k8s ingress) or the juju unit's
(gunicorn-k8s) assigned IP, and you'll be presented with a screen
that details all the environment variables used by the deployed docker image.
"""
The actual hostname in this case would be `gunicorn-k8s` not `gunicorn`.

File: config.yaml [hunk 4]
```diff
@@ -13,7 +13,9 @@ options:
     default: ''
   external_hostname:
     type: string
-    description: "External hostname this gunicorn app should respond to (required)."
+    description: >
+      External hostname this gunicorn app should respond to (required).
```
Comment (arturo-seijas) [inline config.yaml:17]: Shouldn't we remove "required" here?

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by mthaddon — APPROVED** [review]
LGTM, thx

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**